### ISSUE TITLE
Display the gene expression annotation from Canto

### ISSUE THREAD

**Issue by kimrutherford** [pombase/website#1637]
Depends on: pombase/pombase-chado#802

**Comment by ValWood** [comment]
This has "discuss" label. What do you need to know?

**Comment by kimrutherford** [comment]
I don't know where and how the annotation should be displayed.  Does it just go in the Qualitative gene expression section?  Maybe that's happening already?

(I admit to not having thought this through)

**Comment by ValWood** [comment]
Yes, it's exactly the same as the existing 
Quantitative and qualitative expression sections, so maybe it already appears. I can't check because I don't remember which Canto sessions I used them in (or even if I have yet??)

**Comment by mah11** [comment]
I did some expression annotations for PMID:16420355 (session 5e96383ce64d7245), but they don't show up on web pages yet. Instead, there are messages in the curation-tool-data-load-output log that say "can't handle data of type wt_rna_expression". I think I've done some others as well, but would really have to dig to recall which paper(s) -- but there are more messages in the log than there are annotations from PMID:16420355.

Note that the log messages don't include session IDs, so we can't track them down ...

**Comment by kimrutherford** [comment]
>  "can't handle data of type wt_rna_expression".

Thanks Midori.  That confirms that the Canto gene-ex annotation definitely isn't being loaded into Chado.

I've had a look at how the existing gene-ex annotation is stored in Chado and it's going to need updating.  The annotation doesn't use the PomGeneEx ontology.  I'll change that so that it matches the annotation from Canto.

**Comment by kimrutherford** [comment]
> I'll change that so that it matches the annotation from Canto.

Turns out that it's not that easy.  I'll need ontology help for this, probably on a call.

I tested loading the Canto gene-ex annotation into Chado and ended up with post-composed terms like:
```
decreased [during] cellular response to oxidative stress
absent [during] single-celled organism vegetative growth phase
```

The "decreased" and "absent" are what we have in the PomGeneEx ontology, then we add the extension (if any).

We need a way to include RNA and protein in these terms because currently that rather important detail is lost when loading.

**Comment by mah11** [comment]
Will it help if I make "RNA level blah" and "protein level blah" versions of each PomGeneEx term? It wouldn't be a lot of work.

**Comment by kimrutherford** [comment]
> Will it help if I make "RNA level blah" and "protein level blah" versions of each PomGeneEx term? It wouldn't be a lot of work.

That's probably the most straight forward fix.  We'll need to coordinate so that I change the existing Canto gene-ex annotation to use the new terms at the same time.  Let's have a go in January.

**Comment by mah11** [comment]
OK, I have an updated ontology file ready to commit, with RNA and protein subclasses for each qualifier, e.g.

    [Term]
    id: PomGeneEx:0000011
    name: RNA level increased
    def: "A gene expression qualifier that indicates that an RNA is present at a higher level under one set of conditions or at one indicated time than otherwise." [PomBase:mah]
    synonym: "RNA expression increased" EXACT [PomBase:mah]
    synonym: "increased RNA level" EXACT [PomBase:mah]
    is_a: PomGeneEx:0000002
    is_a: PomGeneEx:0000009

Let me know when you're ready for me to commit it.

**Comment by kimrutherford** [comment]
> Let me know when you're ready for me to commit it.

Thanks.

Could you send me a copy?  I'll test the changes on my desktop first and the commit everything once it's all OK.

**Comment by mah11** [comment]
emailed so I don't have to faff with the file extension

**Comment by kimrutherford** [comment]
Thanks for the file.

Would there be any problem with adding a `namespace: ` to the RNA and protein terms?  It would make Canto configuration much simpler if the RNA and protein terms were in different namespaces.  Nothing else needs changing.  I'll email my proposed changes.

```
[Term]
id: PomGeneEx:0000011
name: RNA level increased
namespace: PomGeneExRNA
def: "A gene expression qualifier that indicates that an RNA is present at a higher level under one set of conditions or at one indicated time than otherwise." [PomBase:mah]
synonym: "RNA expression increased" EXACT [PomBase:mah]
synonym: "increased RNA level" EXACT [PomBase:mah]
is_a: PomGeneEx:0000002
is_a: PomGeneEx:0000009
```

**Comment by mah11** [comment]
OK; that won't cause me any trouble.

On a tangentially related note, Juan has some data that will require at least one new expression type, and I'm going to adjust some of the qualifier text defs to allow for it. (No other action on this ticket; it'll affect config for bulk files and I think I have that in hand.)

**Comment by kimrutherford** [comment]
I've committed the updated gene-ex OBO file and updated the annotations in Canto to use the new terms.  I've also updated the Canto configuration to use the new terms.

Please let me know if you spot any problems.

The next step for me is to test loading the updated Canto annotations into Chado.

**Comment by mah11** [comment]
I've just checked quickly, but it seems to be fine so far :)

**Comment by kimrutherford** [comment]
I think this done now.  The gene expression annotation from Canto, the contig files and from the HTP files looks good.  Please let me know if you spot any problems.

As a side effect, we also now also have term pages for the RNA/protein level terms:
   https://www.pombase.org/term/PomGeneEx:0000019

There are no links to those pages yet.  We can add them if you think the term pages are useful.

**Comment by ValWood** [comment]
I think this could be quite a useful page if we organised by "during"  in the display.

**Comment by kimrutherford** [comment]
> I think this could be quite a useful page if we organised by "during" in the display.

Is filtering enough?  If you're in the detailed view you can filter by "during":

![during-filter-1](https://user-images.githubusercontent.com/90474/105660932-082af000-5f31-11eb-9889-570a37e4a842.png)

**Comment by kimrutherford** [comment]
The gene expression sections looks a bit different to the other annotation sections.  Now that the qualitative gene expression annotation has more terms it could use the normal annotation table code.  Do you want to change?

Here is what it would look like:

![new-look-gene-ex-1](https://user-images.githubusercontent.com/90474/105661302-f9910880-5f31-11eb-9858-3300b5e99091.png)
![new-look-gene-ex-2](https://user-images.githubusercontent.com/90474/105661306-fc8bf900-5f31-11eb-8d60-df7e1f58242b.png)

**Comment by ValWood** [comment]
> it could use the normal annotation table code. Do you want to change?

I don't see why not...

**Comment by mah11** [comment]
Seconded - might as well change to the table code that all the cool kids use.

**Comment by ValWood** [comment]
> Is filtering enough? If you're in the detailed view you can filter by "during":

Filtering would work. Because "during" is visible by default, could the  "during" filter be visible in the summary view?

**Comment by kimrutherford** [comment]
> I don't see why not...

Done!

> Because "during" is visible by default, could the "during" filter be visible in the summary view?

Yep.  It should be just a configuration change.  I'll give it a go.

**Comment by ValWood** [comment]
> Yep. It should be just a configuration change. I'll give it a go.

and on the gene pages too, this filter  could be visible by default

**Comment by kimrutherford** [comment]
> Because "during" is visible by default, could the "during" filter be visible in the summary view?

I tested it and it didn't work sensibly.  If I select "mitotic cell cycle" it shows only terms with that extension (which is good) but shows all the extensions for that term (which is bad):
![new-look-gene-ex-4](https://user-images.githubusercontent.com/90474/105706458-6e863180-5f76-11eb-832b-e8ee139e69e2.png)

That's not too bad on the gene pages but is useless on the term pages.  I'll work on fix.

**Comment by kimrutherford** [comment]
> I tested it and it didn't work sensibly.

I've made a separate issue (#1667) for the filtering bug so I can close this.
